src/Modal.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import ReactModal from 'react-modal';
+
+let rootSelector;
+
+export const setAppElement = (selector) => {
+    rootSelector = selector;
+    ReactModal.setAppElement(selector);
+};
+
+export class Modal extends Component {
+    rootElement = null;
+    previousScrollValue = null;
+    prevStyleValues = {};
+
+    componentDidMount() {
+        const { isOpen } = this.props;
+
+        this.rootElement = document.querySelector(rootSelector);
+
+        if (isOpen) {
+            const newStyleValues = this.computeElementValues();
+
+            this.rootElement.style.top = newStyleValues.top;
+            this.rootElement.style.left = newStyleValues.left;
+            this.rootElement.style.right = newStyleValues.right;
+            this.rootElement.style.position = newStyleValues.position;
+        }
+    }
+
+    getSnapshotBeforeUpdate(prevProps) {
+        const { isOpen: wasOpen } = prevProps;
+        const { isOpen } = this.props;
+
+        if (!wasOpen && isOpen) {
+            return this.computeElementValues();
+        }
+
+        if (wasOpen && !isOpen) {
+            return {
+                top: this.prevStyleValues.top,
+                left: this.prevStyleValues.left,
+                right: this.prevStyleValues.right,
+                bottom: this.prevStyleValues.bottom,
+                position: this.prevStyleValues.position,
+            };
+        }
+
+        return null;
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        const { isOpen: wasOpen } = prevProps;
+        const { isOpen } = this.props;
+
+        if (wasOpen !== isOpen) {
+            this.rootElement.style.top = snapshot.top;
+            this.rootElement.style.left = snapshot.left;
+            this.rootElement.style.right = snapshot.right;
+            this.rootElement.style.position = snapshot.position;
+        }
+
+        if (wasOpen && !isOpen && this.previousScrollValue !== null) {
+            window.scroll(0, this.previousScrollValue);
+        }
+    }
+
+    render() {
+        return <ReactModal { ...this.props } />;
+    }
+
+    computeElementValues() {
```

Comment:
This method is strange :/

We are updating `this.prevStyleValues` here and returning almost always the same thing (except for the scroll value).

I think it would be better two separated methods:
- one to update `this.prevStyleValues`
- one to return the "fixed" related values

Reply:
I don't see the need for creating two methods since they'll always be used together. What I do have problem with it's the name of this method as I think it doesn't really describe what it does since it not only computes the new element styles but also saves the current values to use later.

I have a few suggestions for this method, please provide feedback if changing to one of these names would help:
- `calculateElementStyles`
- `calculateAndSaveElementStyles`
- ... a variation of these words? [`calculate`, `elementStyles`, ?]

I'm not sure what's the best name here..

Reply:
Changed it to `calculateElementStyles` :+1: